### PR TITLE
DMP-1511 - Error - 403 - No permission at courthouse for download audio request

### PR DESCRIPTION
**Pull request by joshw-hmcts**
### Jira link (if applicable)

[DMP-1511](https://tools.hmcts.net/jira/browse/DMP-1511)

### Change description ###

Show an error screen when attempting to request audio from a courthouse that the user doesn't have permission for.

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

**Review by DavidTMann — CHANGES_REQUESTED** [review]
Screenshot 2023-11-15 at 13 48 50

We need to hide reporting restrictions on this screen if error occurs

Also please add functional tests in cypress request-audio-spec.cy for checking error message for audio request against that hearing

Could do with a unit test to confirm 403 is not being handled globally by ensuring navigateByUrl has not been called (similar example line 342 search component spec)

There is also a bug if you request audio for hearing 3 it shows error as expected, if you go back (don't refresh) and go to hearing 1 and request it will show same error

File: src/app/components/hearing/hearing.component.ts [hunk 1]
```diff
@@ -103,6 +105,8 @@ export class HearingComponent implements OnInit {
   hearing$ = this.caseService.getHearingById(this.caseId, this.hearingId);
   audio$ = this.hearingService.getAudio(this.hearingId);
   events$ = this.hearingService.getEvents(this.hearingId);
+  private appConfigService = inject(AppConfigService);
```
Comment (davet1985) [inline src/app/components/hearing/hearing.component.ts:108]: I think this should be moved to the top with the other injected services

File: src/app/components/hearing/hearing.component.html [hunk 2]
```diff
@@ -12,9 +12,19 @@
...
+  <ng-container *ngIf="state === 'OrderFailure'">
+    <h1 class="govuk-heading-xl govuk-!-margin-bottom-8">
+      You do not have permission to get audio for this courthouse.
+    </h1>
+    <p class="govuk-body">
+      If you believe you should have permission,
+      <a href="mailto:{{ support?.emailAddress }}" class="govuk-link">contact {{ support?.name }}</a
+      >.
+    </p>
+  </ng-container>
```
Comment (davet1985) [inline src/app/components/hearing/hearing.component.html:27]: Could this use `<app-forbidden></app-forbidden>` with the addition of a new `Input` property to override the default heading?

File: cypress/e2e/functional/request-audio-spec.cy.js [hunk 3]
```diff
@@ -55,13 +55,52 @@ describe('Request audio', () => {
...
 
+  it('should show an error screen if a 403 is returned', () => {
+    cy.get('#hearingsTable').should('contain', '1 Oct 2023');
+    cy.get('#hearingsTable a').contains('1 Oct 2023').click();
+    cy.get('.button').contains('Get Audio').click();
+    cy.get('.govuk-error-summary').should('contain', 'You must include a start time for your audio recording');
+    cy.get('.govuk-error-summary').should('contain', 'You must include an end time for your audio recording');
+    cy.get('.govuk-error-summary').should('contain', 'You must select a request type');
+
+    cy.get(':nth-child(2) > [scope="row"] > .govuk-checkboxes > .govuk-checkboxes__item > #\\31 ').click();
```
Comment (davet1985) [inline cypress/e2e/functional/request-audio-spec.cy.js:66]: Feel free to add a `id` attribite or extra `class` to help target elements.

**Review by davet1985 — APPROVED** [review]

**Pull request merged**